smart-contracts/test/Lock/disableLock.js

```diff
@@ -0,0 +1,99 @@
+const Units = require('ethereumjs-units')
+const Web3Utils = require('web3-utils')
+const BigNumber = require('bignumber.js')
+
+const deployLocks = require('../helpers/deployLocks')
+const shouldFail = require('../helpers/shouldFail')
+const Unlock = artifacts.require('../Unlock.sol')
+
+let unlock, locks, ID
+
+contract('Lock', accounts => {
+  let lock
+
+  before(async () => {
+    unlock = await Unlock.deployed()
+    locks = await deployLocks(unlock)
+    lock = locks['FIRST']
+    await lock.purchaseFor(accounts[1], Web3Utils.toHex('Julien'), {
+      value: Units.convert('0.01', 'eth', 'wei')
+    })
+    ID = new BigNumber(await lock.getTokenIdFor(accounts[1])).toFixed()
+  })
+
+  describe('disableLock', () => {
+    it('should fail if called by the wrong account', async () => {
+      await shouldFail(lock.disableLock({ from: accounts[1] }), '')
+    })
+
+    it('should fail if called before the lock is disabled', async () => {
+      await shouldFail(
+        lock.destroyLock(),
+        'Not allowed to delete an active lock'
+      )
+    })
+
+    describe('when the lock has been disabled', () => {
```

Comment:
Maybe add a test to explicitly confirm we can still read information such as `getHasValidKey` while the Lock is disabled.

Reply:
done.